mindmeld/components/entity_recognizer.py

```diff
@@ -156,7 +143,13 @@ def dump(self, model_path, incremental_model_path=None):
         logger.info(
             "Saving entity classifier: domain=%r, intent=%r", self.domain, self.intent
         )
-        super().dump(model_path, incremental_model_path)
+        metadata = {
```

Comment:
A suggestion: instead of initializing different metadata dictionaries at various points in the code: in individual classifiers (with these keys), in the base model (with the model and model config keys) etc., a more ideal solution might be to create a Metadata/Metadata manager class separately and define all possible attributes for the class explicitly there. 

* Instance(s) can then be utilized at different points of this refactor and attributes tracked more easily in one place
* It would be cleaner to update instances through classmethods instead of standalone dictionary updates
* validation for metadata attributes can be done (using marshmallow).